### PR TITLE
Data base test

### PR DESCRIPTION
**Pull request by hamzaouadia**
# Test Database Infrastructure Setup

## Summary
Adds Docker-based PostgreSQL infrastructure for running integration tests in isolation from the development database.

## What's Added
- `docker-compose.test.yml` - PostgreSQL 15 container on port 5433
- `.env.test` - Test database connection strings
- `scripts/setup-test-db.sh` - Automated setup script
- `scripts/init-test-db.sh` - Alternative setup with seed support

## Quick Start
```bash
chmod +x scripts/setup-test-db.sh
./scripts/setup-test-db.sh
pnpm test

### CODE REVIEW

File: packages/api/src/tests/fixtures/index.ts [hunk 1]
```diff
@@ -0,0 +1,306 @@
+/**
+ * Test fixtures for seeding the test database
```
Comment (BrightonMboya) [inline packages/api/src/tests/fixtures/index.ts]: For seeding here lets use the pattern I shared on the other PRs. Also if we are deleting the values on every tests, that means the db will be empty every time we start working no?

File: packages/api/src/tests/routers/auth.test.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { describe, it, expect } from "vitest";
+import { createCaller } from "../../root";
```
Comment (BrightonMboya) [inline packages/api/src/tests/routers/auth.test.ts]: Do we need these files under routers/** to set up a development database?

File: packages/api/src/tests/fixtures/index.ts (hunk 1 above)
Comment (BrightonMboya) [reply packages/api/src/tests/fixtures/index.ts]: or maybe the name should be test-database to not confuse us wdyt?

File: scripts/setup-test-db.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+echo "⏳ Waiting for PostgreSQL to be ready..."
+DB_CONTAINER_ID="$(docker compose -f "$ROOT_DIR/docker-compose.test.yml" ps -q db-test)"
+until docker exec "$DB_CONTAINER_ID" pg_isready -U postgres >/dev/null 2>&1; do
+  sleep 1
+done
+
+echo "✅ PostgreSQL is ready"
+
+# Export test database URL
+export DATABASE_URL="postgresql://postgres:password@localhost:5433/kilimo_test"
```
Comment (BrightonMboya) [inline scripts/setup-test-db.sh]: is this correct? both urls looks the same to me

File: scripts/setup-test-db.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+export DIRECT_URL="postgresql://postgres:password@localhost:5433/kilimo_test"
+
+# Push Prisma schema to test database
+echo "🔄 Pushing Prisma schema to test database..."
+cd "$ROOT_DIR/packages/db"
+DATABASE_URL="postgresql://postgres:password@localhost:5433/kilimo_test" DIRECT_URL="postgresql://postgres:password@localhost:5433/kilimo_test" pnpm exec prisma db push --skip-generate
```
Comment (BrightonMboya) [inline scripts/setup-test-db.sh]: also i think these should come from env file no?

**Review by BrightonMboya — CHANGES_REQUESTED** [review]

File: packages/api/src/tests/fixtures/index.ts (hunk 1 above)
Comment (hamzaouadia) [reply packages/api/src/tests/fixtures/index.ts]: well as i said in the other PR i can add a seed integration based on the Prisma documentation

Comment (hamzaouadia) [reply packages/api/src/tests/fixtures/index.ts]: but for now we need to keep the fixtures folder

File: packages/api/src/tests/routers/auth.test.ts (hunk 2 above)
Comment (hamzaouadia) [reply packages/api/src/tests/routers/auth.test.ts]: I can update the test files to import from the package entry (e.g. import { createCaller } from "@kilimo/api"). Useful if we want a single canonical import path rather than relative paths.

File: scripts/setup-test-db.sh (hunk 4 above)
Comment (hamzaouadia) [reply scripts/setup-test-db.sh]: Yes if we thing that its a type of credential. no because it just for a local test development

File: scripts/setup-test-db.sh (hunk 3 above)
Comment (hamzaouadia) [reply scripts/setup-test-db.sh]: both urls?

**Review by BrightonMboya — COMMENTED** [review]
Okay nice idea on this one, however let me challenge your pov

We are using supabase as our provider, that means they have supabase locally setup guide. How about we just follow their guides on having supabse set up locally ? that way we are sure that our development env mirros 1:1 that what we are having on prod

**Comment by BrightonMboya** [comment]
@hamzaouadia do we still need this PR now you are working on the supabase local setup?
